### PR TITLE
Feature recommendation 

### PR DESCRIPTION
**Pull request by flyinghermit**
This PR adds:
- `RecommendFeature` type which holds feature recommendation status `'NOTIFY' | 'DONE'` for given feature.
   ```
     export type RecommendFeature = {
        TrustedDevices: RecommendationStatus;
     };
    
     export type RecommendationStatus = 'NOTIFY' | 'DONE';
    ```

- adds a local storage item `grv_recommend_feature` to track `RecommendFeature`.
- shows "red dot" in side navigation if notification status for given feature is `'NOTIFY'` and cluster already has at least one resource connected, which is checked with `const onboard = localStorage.getOnboardDiscover(); onboard.hasResource = True`
- Side navigation menu is selected based on navigation item title. To avoid cases where title is updated in the feature and the string comparison breaks, I have added `enum NavTitles` and updated `TeleportFeatureNavigationItem`
    ```
    TeleportFeatureNavigationItem {
       title: NavTitles;
       ....
   }
     ``` 
- currently supports  trusted devices but the mechanism is designed to be used for other features as well (since we have ongoing plan to highlight other existing enterprise features)

See `e` counterpart for more description - https://github.com/gravitational/teleport.e/pull/1912

### CODE REVIEW

**Comment by sshahcodes** [comment]
"red dot" in side nav:
Screenshot 2023-08-01 at 7 18 44 PM

File: web/packages/teleport/src/types.ts [hunk 1]
```diff
@@ -29,13 +29,59 @@ export interface Context {
...
 export interface TeleportFeatureNavigationItem {
-  title: string;
+  title: NavTitles;
   icon: React.ReactNode;
   exact?: boolean;
   getLink?(clusterId: string): string;
   isExternalLink?: boolean;
 }
 
+export type NavTitle = `${NavTitles}`;
```
Comment (ryanclark) [inline web/packages/teleport/src/types.ts:39]: What's this used for?

File: web/packages/teleport/src/types.ts [hunk 2]
```diff
@@ -118,3 +164,10 @@ export type LockedFeatures = {
   premiumSupport: boolean;
   trustedDevices: boolean;
 };
+
+// RecommendFeature is used for recommending features if its usage status is zero.
+export type RecommendFeature = {
+  TrustedDevices: RecommendationStatus;
+};
+
+export type RecommendationStatus = 'NOTIFY' | 'DONE';
```
Comment (ryanclark) [inline web/packages/teleport/src/types.ts:173]: Can this be an enum?

Comment (sshahcodes) [reply web/packages/teleport/src/types.ts]: Updated this to enum in https://github.com/gravitational/teleport/pull/29757/commits/007fcdb19d5d45a2e664b0aa9208dbe9b446e9ee

File: web/packages/teleport/src/types.ts (hunk 1 above)
Comment (sshahcodes) [reply web/packages/teleport/src/types.ts]: Thanks for pointing out, I defined it to be used as a type built from enum but ended up not using it anywhere. Removed in https://github.com/gravitational/teleport/commit/007fcdb19d5d45a2e664b0aa9208dbe9b446e9ee

File: web/packages/teleport/src/Navigation/NavigationItem.tsx [hunk 3]
```diff
@@ -154,6 +158,26 @@ export function NavigationItem(props: NavigationItemProps) {
     []
   );
 
+  // showRedDot returns red dot component if the feature recommendation state is 'NOTIFY'
+  function showRedDot(featureName: NavTitle): JSX.Element {
```
Comment (kimlisa) [inline web/packages/teleport/src/Navigation/NavigationItem.tsx:162]: `showRedDot` sounds like a flag
```suggestion
  function renderHighlightFeature(featureName: NavTitle): JSX.Element {
```
or something like that

**Review by kimlisa — APPROVED** [review]

**Review by ryanclark — APPROVED** [review]

**Pull request merged**